File: App/AI-service/models/AssessmentSummaryDTO.py
from pydantic import BaseModel, Field
from typing import Optional

class AssessmentSummaryDTO(BaseModel):
    finalScore: float = Field(..., description="Computed questionnaire score")
    severityLabel: str = Field(..., description="Severity classification")

    clinicalDescription: Optional[str] = Field(
        None,
        description="Clinical interpretation from scoring rules"
    )

    minScore: Optional[float] = None
    maxScore: Optional[float] = None
    
    class Config:
        extra = "forbid"

File: App/Questionnaire/src/main/java/com/example/mentalhealth/aiService/DTO/AssessmentSummaryDTO.java
package com.example.mentalhealth.aiService.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AssessmentSummaryDTO {

    private Double finalScore;              // REQUIRED
    private String severityLabel;            // REQUIRED
    private String clinicalDescription;      // from scoring rules
    private Double minScore;                 // optional but recommended
    private Double maxScore;                 // optional but recommended
}